packages/library-base/src/commonMain/kotlin/io/realm/kotlin/notifications/MapChange.kt

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2023 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.kotlin.notifications
+
+import io.realm.kotlin.types.RealmDictionary
+import io.realm.kotlin.types.RealmMap
+
+/**
+ * This sealed interface describes the possible changes that can happen to a [RealmMap].
+ *
+ * The states are represented by the specific subclasses [InitialMap], [UpdatedMap] and
+ * [DeletedMap]. When the map is deleted an empty map is emitted instead of `null`.
+ *
+ * Since maps do not expose indices your UI components will have to manually handle updates:
+ *
+ * ```
+ * person.addresses.asFlow()
+ *   .collect { mapChange: MapChange<String, Address> ->
+ *       handleChange(mapChange.map)
+ *   }
+ * ```
+ */
+public sealed interface MapChange<K, V> {
+    public val map: RealmMap<K, V>
+}
+
+/**
+ * Convenience alias for [RealmDictionary] changes. It represents a [MapChange] of `<String, V>`.
+ */
+public typealias DictionaryChange<V> = MapChange<String, V>
```

Comment:
Do we need these? Since we are setting these up anyway in our `asFlow` method. I'm not sure what value they bring?

Reply:
Same for the other ones below.

Reply:
I don't think we need them per se. I thought they might be a more convenient way to establish boundaries to all things dictionary just as I did for `RealmDictionaryEntrySet` and `RealmDictionaryMutableEntry`.